cql3/statements/alter_table_statement.cc

```diff
@@ -438,15 +452,26 @@ alter_table_statement::raw_statement::prepare(data_dictionary::database db, cql_
     }
 
     auto ctx = get_prepare_context();
+    auto prepared_attrs = _attrs->prepare(db, keyspace(), column_family());
+    prepared_attrs->fill_prepare_context(ctx);
+
+    auto bound_terms = ctx.bound_variables_size();
+
+    if (bound_terms > 0 && !s) {
+        throw exceptions::invalid_request_exception("Cannot prepare a statement with parameter markers with non-existing schema");
+    }
 
     return std::make_unique<prepared_statement>(::make_shared<alter_table_statement>(
-        ctx.bound_variables_size(),
-        std::move(*_cf_name),
-        _type,
-        std::move(_column_changes),
-        std::move(_properties),
-        std::move(_renames)
-    ));
+                std::move(bound_terms),
+                std::move(*_cf_name),
+                _type,
+                std::move(_column_changes),
+                std::move(_properties),
+                std::move(_renames),
```

Comment:
Please don't change the indentation if you already introduce some logic changes in the same commit. While it's not a big readability concern here, in general doing this makes review harder.

I recommend doing a short self-review, i.e. looking at how the commits currently look like before pushing a new version to the remote branch. It helps catch issues like this.

Reply:
I keep forgetting about it :grimacing: